src/main/java/com/launchdarkly/eventsource/UnsuccessfulResponseException.java

```diff
@@ -1,27 +1,64 @@
 package com.launchdarkly.eventsource;
 
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.Response;
+
 /**
  * Exception class that means the remote server returned an HTTP error.
  */
 @SuppressWarnings("serial")
 public class UnsuccessfulResponseException extends Exception {
 
-  private final int code;
+  private final Response response;
 
   /**
    * Constructs an exception instance.
    * @param code the HTTP status
    */
+  @Deprecated
   public UnsuccessfulResponseException(int code) {
-    super("Unsuccessful response code received from stream: " + code);
-    this.code = code;
+    this(buildSurrogateResponse(code));
```

Comment:
Actually, I also realized that the whole PR didn't work.

I think due to the async fact that the whole response is not available yet when `UnsuccessfulResponseException` is created within the library's code (unlike the error code).
In other words, there are more changes needed (even if this constructor change is reverted).

So far, we happily used the library without this change anyway - @eli-darkly , I'm OK if it is rejected (when I need it badly, I'll propose proper change).

Reply:
Hmm. Can you say more about what specifically was not working? I would expect that trying to read the _body_ would not work, since other code might already have read the body - that's just an expected issue whenever you share a Response around. But I would have expected the headers to be available.